internal/controller/promotion/argocd.go

```diff
@@ -215,9 +219,46 @@ func (a *argoCDMechanism) doSingleUpdate(
 	}
 	logging.LoggerFromContext(ctx).WithField("app", app.Name).
 		Debug("patched Argo CD Application")
+
+	a.logAppEventFn(ctx, app, "Promotion", "Promotion triggered a sync of this Application resource.")
+
 	return nil
 }
 
+func (a *argoCDMechanism) logAppEvent(ctx context.Context, app *argocd.Application, reason, message string) {
+	logger := logging.LoggerFromContext(ctx)
+
+	t := metav1.Time{Time: time.Now()}
+	event := corev1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%v.%x", app.Name, t.UnixNano()),
```

Comment:
As we are working with a shim of the Argo CD API, we would have to introduce more structures to be able to use the same annotations as being set in https://github.com/argoproj/argo-cd/blob/820f4d861a7789f299143ed89816001091abf923/util/argo/audit_logger.go#L119-L120

In addition, because we are working with a partial spec. I was not sure if we should include it as being done in: https://github.com/argoproj/argo-cd/blob/820f4d861a7789f299143ed89816001091abf923/util/argo/audit_logger.go#L125

Reply:
I'm checking with some folks, I think for completeness and the way we parse these events (outside of Kargo), we may end up wanting the other annotations.

Reply:
@jessesuen I think we need to make a decision soon on if this gets into v0.6.0 or gets deferred.

Reply:
Confirmed that the annotations are not needed.